web/client/reducers/annotations.js

```diff
@@ -743,6 +754,21 @@ function annotations(state = {validationErrors: {}}, action) {
     }
     case START_DRAWING:
         return {...state, config: {...state.config, geodesic: get(action.options, 'geodesic', false)}};
+    case VALIDATE_FEATURE:
+
+        let updatedFeatures = state.editing.features;
```

Comment:
doing a map and not assigning to anything is not correct and mutating things like this are not wrong per se, but we try to use a different way
```suggestion
        let updatedFeatures = state.editing.features.map((f) => {
             const isValidFeature = validateFeature({
                properties: f.properties,
                components: getComponents(f.geometry),
                type: f.geometry.type
            });
        return {...f, properties: {...f.properties, isValidFeature }}
        });
```
OR
```suggestion
        let updatedFeatures = state.editing.features.map((f) => {
             const isValidFeature = validateFeature({
                properties: f.properties,
                components: getComponents(f.geometry),
                type: f.geometry.type
            });
            return set(`properties.isValidFeature`, isValidFeature, f);
        });
```
with set we do not mutate the original object but we return a new object mutated

Reply:
Thanks, corrected!